### PR TITLE
Add GitHub workflow to create release and upload to PyPI when version number is bumped

### PR DESCRIPTION
**Pull request by lfdebrux**
This PR adds two workflows to replace [our current Jenkins job to tag new versions of this package](https://ci.marketplace.team/job/tag-dmtest-utils/), and also adds a new [`digitalmarketplace-test-utils` PyPI package](https://pypi.org/project/digitalmarketplace-test-utils/).

The hope is that by uploading packages to PyPI we will be able to get Dependabot to manage our library versions across our apps, and save us a lot of work updating packages.

### CODE REVIEW

**Comment by lfdebrux** [comment]
I think the only way to test this is to merge it and try making a release, so I'm going to merge this once it gets an approval and if we decide later we don't like it I will revert the changes.

**Review by gidsg — APPROVED** [review]
👍 📦

**Comment by lfdebrux** [comment]
🤞

**Pull request merged**
